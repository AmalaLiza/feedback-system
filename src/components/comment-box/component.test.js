import React from "react";
import { shallow } from "enzyme";
import CommentBox from "./component";
import renderer from "react-test-renderer";

const props = {
	comment: {
		id: 1,
		name: "Sheldon Cooper",
		email: "jonaskahnwald@dark.anotherworld",
		comment: "The end is the beginning, and the beginning is the end.",
		rating: 5,
	},
};

function setup() {
	const wrapper = shallow(<CommentBox {...props} />);
	return { wrapper, props };
}

describe("CommentBox Component Test Suite", () => {
	it("renders correctly", () => {
		const tree = renderer.create(<CommentBox {...props} />).toJSON();
		expect(tree).toMatchSnapshot();
	});

	it("Should have an li", () => {
		const { wrapper } = setup();
		expect(wrapper.find("li").exists()).toBe(true);
	});
});
