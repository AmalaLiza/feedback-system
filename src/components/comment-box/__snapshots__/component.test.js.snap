// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CommentBox Component Test Suite renders correctly 1`] = `
<li
  className="commentSection"
>
  <header
    className="header"
  >
    <div
      className="avatar"
    >
      S
    </div>
    <span
      className="name"
    >
      Sheldon Cooper
    </span>
    <div
      className="metadata"
    >
      <div
        className="commentDate"
      >
        Invalid Date
      </div>
      <div
        className="commentDate"
      >
        <img
          className="star"
          src={
            Object {
              "process": [Function],
            }
          }
        />
        5
      </div>
    </div>
  </header>
  <div
    className="commentContent"
  >
    <p
      className="email"
    >
      <span
        className="emailLabel"
      >
        E-mail 
      </span>
      <span>
        jonaskahnwald@dark.anotherworld
      </span>
    </p>
    <p>
      The end is the beginning, and the beginning is the end.
    </p>
  </div>
</li>
`;
